src/editor/components/modals/file-modal/file-list-item.jsx

```diff
@@ -0,0 +1,219 @@
+import React from "react";
+import PropTypes from "prop-types";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import styled from "react-emotion";
+
+import { TextButton } from "../../../../shared/components/buttons";
+import {
+  ListItem,
+  LIST_BORDER,
+  LIST_BORDER_RADIUS
+} from "../../../../server/components/list";
+import THEME from "../../../../shared/theme";
+import { fileShape } from "./propShapes";
+import { defaultSpacing } from "./styles";
+
+const circularProgressSize = "20px";
+
+const FileListItem = styled(ListItem)`
+  align-content: center;
+  align-items: center;
+  display: grid;
+  grid-auto-flow: column;
+  grid-gap: ${defaultSpacing};
+  justify-content: space-between;
+  min-height: ${circularProgressSize};
+
+  &.error {
+    background-color: #f4cccc;
+  }
+
+  &.deleted {
+    border: 0;
+    height: 0;
+    min-height: 0;
+    opacity: 0;
+    padding: 0;
+
+    transition: border 250ms step-end, height 250ms step-end,
+      min-height 250ms step-end, padding 250ms step-end, opacity 250ms;
+  }
```

Comment:
Hey @openjck – I love this approach you started with this kind of transition. I'm not 100% sold on the idea of keeping the elements (vs. unmounting), but I don't think in practice it matters much. That this isn't easy is to me a real pain when using React. At any rate, the above changes here actually also animate the collapsing of the element, so it's a bit easier for the eye to track what happens when you delete. Try these suggestions out and let me know what you think.

Reply:
I don't think it's possible to animate them without keeping them in the DOM. Transitions can't be applied to elements removed from the DOM or set to `display: none`. I'll test out your animations and let you know what I think!

Reply:
I like this. I made a couple of small tweaks but nothing huge.